source/syntax/SyntaxVisitor.cpp

```diff
@@ -54,6 +50,8 @@ struct CloneVisitor {
             if (cloned->kind != SyntaxKind::SyntaxList && cloned->kind != SyntaxKind::SeparatedList)
```

Comment:
There's no reason to have this backfill function at all anymore. You call it on all paths through the loop, so index will always be zero on the first call and it will do nothing but emplace() on the listBuffer, and every subsequent call will early return. Just make listBuffer not optional and get rid of this lambda function.